crates/nu-cli/src/commands/classified/external.rs

```diff
@@ -333,16 +333,27 @@ fn spawn(
                                 }
                             }
                         },
-                        Err(_) => {
-                            let _ = stdout_read_tx.send(Ok(Value {
-                                value: UntaggedValue::Error(ShellError::labeled_error(
-                                    "Unable to read from stdout.",
-                                    "unable to read from stdout",
-                                    &stdout_name_tag,
-                                )),
-                                tag: stdout_name_tag.clone(),
-                            }));
-                            break;
+                        Err(e) => {
+                            // If there's an exit status, it makes sense that we may error when
+                            // trying to read from its stdout pipe (likely been closed). In that
+                            // case, don't emit an error.
+                            let should_error = match child.wait() {
+                                Ok(exit_status) => !exit_status.success(),
+                                Err(_) => true,
+                            };
+
+                            if should_error {
+                                let _ = stdout_read_tx.send(Ok(Value {
+                                    value: UntaggedValue::Error(ShellError::labeled_error(
+                                        format!("Unable to read from stdout ({})", e),
```

Comment:
I also decided that it would probably be good to include the error message in here, but can revert that 🙂

Reply:
Sounds good.  In the future, we could move to `codespan` which gives us more access to adding additional notes and whatnot to errors